clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -345,25 +347,26 @@ class DiagnoseHLSLAvailability
 
   // Helper methods for dealing with current stage context / environment
   void SetShaderStageContext(HLSLShaderAttr::ShaderType ShaderType) {
+    static_assert(HLSLShaderAttr::ShaderTypeMaxValue < 31,
+                  "ShaderType is too big for this bitmap");
     assert((((unsigned)1) << (unsigned)ShaderType) != 0 &&
+           (((unsigned)1) << (unsigned)ShaderType) != 31 &&
```

Comment:
`1 << ` any number can't ever be 31 can it?

Maybe it's meant to be something like:

```suggestion
    assert((((unsigned)1) << (unsigned)ShaderType) != 0 &&
           (unsigned)ShaderType) != 31 &&
```